addons/web/models/models.py

```diff
@@ -68,7 +68,7 @@ def _format_web_search_read_results(self, domain, records, offset=0, limit=None,
 
     def web_save(self, vals, specification: dict[str, dict], next_id=None) -> list[dict]:
         if self:
-            self.write(vals)
+            self.with_context(web_save=True).write(vals)
```

Comment:
I guess this side of the hack might also have to be done with some monkey patching. But ideally, I'd rather hope for the opposite condition: was it saved through a wysiwyg editor (and not a `web` view).
See https://github.com/odoo/odoo/blob/master/addons/web_editor/models/ir_ui_view.py#L268. I.e. "clean up the image options unless we actually know about their existence"